src/main/java/sirius/biz/process/Processes.java

```diff
@@ -176,6 +176,20 @@ public String createProcess(@Nullable String type,
         return process.getIdAsString();
     }
 
+    /**
+     * Fetches the currently active process.
+     *
+     * @return the process for which the current thread is executing or an empty optional if no process is active
+     */
+    public Optional<Supplier<Process>> fetchCurrentProcess() {
+        TaskContextAdapter adapter = TaskContext.get().getAdapter();
+        if (adapter instanceof ProcessEnvironment processEnvironment) {
+            return Optional.of(() -> fetchRequiredProcess(processEnvironment.getProcessId()));
```

Comment:
**Question:** Why doesn't this use the `Processes::fetchProcess` variant that given an Optional?

Reply:
We actually return a Supplier<Process> here instead of a Process - as sometimes we only want to check if a process would exist, without actually fetching it. This method sort of hast two purposes, check if a process is present (and if so, fetch it in a second step, if required)...